packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/style/Gradient.kt

```diff
@@ -8,76 +8,58 @@
 package com.facebook.react.uimanager.style
 
 import android.content.Context
-import android.graphics.LinearGradient
 import android.graphics.Rect
 import android.graphics.Shader
 import com.facebook.react.bridge.ColorPropConverter
 import com.facebook.react.bridge.ReadableMap
 import com.facebook.react.bridge.ReadableType
 
-public class Gradient(gradient: ReadableMap?, context: Context) {
+internal class Gradient(gradient: ReadableMap?, context: Context) {
   private enum class GradientType {
     LINEAR_GRADIENT
   }
 
   private val type: GradientType
-  private var startX: Float = 0f
-  private var startY: Float = 0f
-  private var endX: Float = 0f
-  private var endY: Float = 0f
-  private val colors: IntArray
-  private val positions: FloatArray
+  private val linearGradient: LinearGradient
 
   init {
     gradient ?: throw IllegalArgumentException("Gradient cannot be null")
 
     val typeString = gradient.getString("type")
-    type =
-        when (typeString) {
-          "linearGradient" -> GradientType.LINEAR_GRADIENT
-          else -> throw IllegalArgumentException("Unsupported gradient type: $typeString")
-        }
-
-    gradient.getMap("start")?.let { start ->
-      startX = start.getDouble("x").toFloat()
-      startY = start.getDouble("y").toFloat()
+    type = when (typeString) {
+      "linearGradient" -> GradientType.LINEAR_GRADIENT
+      else -> throw IllegalArgumentException("Unsupported gradient type: $typeString")
```

Comment:
Is it guaranteed that process function would not allow through invalid gradients? We otherwise prefer to gracefully no-op on invalid style values instead of throw.

Reply:
yes, in case of invalid syntax the process function will return an empty array, so this error is a very unlikely event